pkg/virt-controller/watch/vmi.go

```diff
@@ -1594,10 +1594,20 @@ func (c *VMIController) resolveControllerRef(namespace string, controllerRef *v1
 			return nil
 		}
 		if !exists {
-			return nil
+			if length := len(controllerRef.Name); length > 20 { // virt-launcher-${vmiName}-mmgpv
+				vmiName := controllerRef.Name[14:length-6]
```

Comment:
IMHO, this is not the right way to go. You probably should not rely on the naming of the virt-launcher pod.